runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -473,8 +466,62 @@ impl<Cfg: Config> API for Module<Cfg> {
 }
 
 impl<Cfg: Config> Module<Cfg> {
-    fn do_evm<C, F>(source: H160, ctx: &C, f: F, estimate_gas: bool) -> Result<Vec<u8>, Error>
+    fn evm_call<C: Context>(
+        ctx: &C,
+        caller: H160,
+        address: H160,
+        value: U256,
+        data: Vec<u8>,
+        estimate_gas: bool,
+    ) -> Result<Vec<u8>, Error> {
+        Self::evm_execute(
+            ctx,
+            caller,
+            |exec, gas_limit| {
+                exec.transact_call(
+                    caller.into(),
+                    address.into(),
+                    value.into(),
+                    data,
+                    gas_limit,
+                    vec![],
+                )
+            },
+            estimate_gas,
+        )
+    }
+
+    fn evm_create<C: Context>(
+        ctx: &C,
+        caller: H160,
+        value: U256,
+        init_code: Vec<u8>,
+        estimate_gas: bool,
+    ) -> Result<Vec<u8>, Error> {
+        Self::evm_execute(
+            ctx,
+            caller,
+            |exec, gas_limit| {
+                let address = exec.create_address(evm::CreateScheme::Legacy {
```

Comment:
This can likely be moved into `exit_reason.is_succeed()` branch? (or does the executor require that this is called before transact_create?)

Reply:
I believe there is a problem with address derivation as it depends on the nonce which can change during execution.

Reply:
Maybe add a comment for future reference.

Reply:
Done.